src/main/java/lotto/domain/Lotto.java

```diff
@@ -0,0 +1,84 @@
+package lotto.domain;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import lotto.domain.numberPicker.NumberPicker;
+import lotto.domain.validator.ParamsValidator;
+import lotto.exception.lotto.LottoNumberCountInvalidException;
+import lotto.exception.lotto.LottoNumberDuplicatedException;
+
+final public class Lotto {
+
+    private static final int LOTTO_NUMBER_COUNT = 6;
+    private static final int PRICE = 1000;
+
+    private final List<Number> numbers;
+
+    private Lotto(final List<Number> numbers) {
+        validateNumbersCount(numbers);
+        validateNumbersNotDuplicated(numbers);
+        this.numbers = numbers.stream()
+                .sorted()
+                .toList();
+    }
+
+    private static void validateNumbersCount(List<Number> numbers) {
+        if (numbers.size() != LOTTO_NUMBER_COUNT) {
+            throw new LottoNumberCountInvalidException(LOTTO_NUMBER_COUNT);
+        }
+    }
+
+    private static void validateNumbersNotDuplicated(List<Number> numbers) {
+        if (hasDuplicatedNumber(numbers)) {
+            throw new LottoNumberDuplicatedException();
+        }
+    }
+
+    private static boolean hasDuplicatedNumber(List<Number> numbers) {
+        return new HashSet<>(numbers).size() != numbers.size();
```

Comment:
네, Stream의 distinct()를 활용한 방법은 어떻게 생각하시나요?

```suggestion
        return numbers.size() != numbers.stream()
                    .distinct()
                    .count();
```

개인적으로 코드가 간결하고 명확하면서도, Stream API의 함수형 스타일로 우아하게 표현된다고 생각합니다!
또 새로운 컬렉션 생성 없이 중복 검증이 가능해서 좋은 것 같아요.

Reply:
요거 좋네요! distinct()의 존재를 까맣게 잊고 있었어요. 이번에 활용해보겠습니다!